src/social/service/comment.service.ts

```diff
@@ -56,10 +60,30 @@ export class CommentService {
     return comment;
   }
 
-  async getCommentsByUserId(userId: string): Promise<Comment[]> {
+  async validateNumber(limitOffset, warning: string): Promise<void> {
+    const limitOffsetInt = Number(limitOffset);
+    if (isNaN(limitOffsetInt) || limitOffsetInt < 0) {
+      throw new BadRequestException(`${warning} must be a positive integer`);
+    }
+  }
+
+  async getCommentsByUserId(userId: string, limit, offset) {
     await this.verifyUserExistence(userId);
 
-    const comments: Comment[] = await this.manager.find(Comment, { userId });
+    let queryStatement = `SELECT * FROM COMMENT WHERE "userId" = '${userId}' `;
```

Comment:
use QueryBuilder to prevent SQL injection
https://typeorm.io/#/select-query-builder/how-to-create-and-use-a-querybuilder

or if still need raw query: https://github.com/typeorm/typeorm/issues/4761

Reply:
Rebased